### PR TITLE
Add onnxoptimizer

### PR DESCRIPTION
**Pull request by neNasko1**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/onnxoptimizer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  number: 0
+
+requirements:
+  build:
+    - cmake
+    - make
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - libgcc
```
Comment (xhochy) [inline recipes/onnxoptimizer/meta.yaml:24]: This should not be here but come in automatically if the C++ compiler thinks, it needs it.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/onnxoptimizer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/onnxoptimizer-{{ version }}.tar.gz
+  sha256: e08b726e0d4577e51e529f36bc324bf11b7cff12852cf3eee081f05c8b8c6f33
+
+build:
+  noarch: python
```
Comment (xhochy) [inline recipes/onnxoptimizer/meta.yaml:13]:
```suggestion
```

File: recipes/onnxoptimizer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - pytest-runner
+  run:
+    - python
+    - onnx
+    - {{ pin_compatible('numpy') }}
```
Comment (xhochy) [inline recipes/onnxoptimizer/meta.yaml:30]: This only works if you have numpy in host.

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by xhochy** [comment]
You probably need to set a C++ standard in CXXFLAGS to fix osx. Changing the SDK is unlikely to fix it.

File: recipes/onnxoptimizer/meta.yaml (hunk 3 above)
Comment (neNasko1) [reply recipes/onnxoptimizer/meta.yaml]: Changed.

**Comment by neNasko1** [comment]
> You probably need to set a C++ standard in CXXFLAGS to fix osx. Changing the SDK is unlikely to fix it.

The problem was that I was specifying the osx development version, not the target platform version. Should be okay now. 

Should be ready for review @xhochy

File: recipes/onnxoptimizer/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    - make
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - python
+    - pip
+    - pytest-runner
+  run:
+    - python
+    - onnx
```
Comment (xhochy) [inline recipes/onnxoptimizer/meta.yaml:29]: You will need a run dependency on the deployment target here. See the conda-forge docs on how to do it.

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by neNasko1** [comment]
@xhochy do you think this needs any more changes or is it ready to be merged?

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by neNasko1** [comment]
@xhochy all the requested changes have been included.

**Pull request merged**